src/couch/priv/couch_js/86/util.cpp

```diff
@@ -26,6 +26,29 @@
 #include "help.h"
 #include "util.h"
 
+const char*
+get_spidermonkey_version() {
+    const char *JAVASCRIPT = "JavaScript-C";
+    int js_len = strlen(JAVASCRIPT);
+
+    // JS_GetImplementationVersion()
+    // returns "JavaScript-CMAJOR.MINOR.PATCH"
+    const char *FULLVERSION = JS_GetImplementationVersion();
+    int fv_len = strlen(FULLVERSION);
+
+    const char* foundJSString = strstr(FULLVERSION,JAVASCRIPT);
+    if (foundJSString != NULL) {
+        //trim off "JavaScript-C",
+        char *buf = (char*) calloc(fv_len - js_len + 1, sizeof(char));
+        strncpy(buf, &FULLVERSION[js_len], fv_len - js_len);
+        buf[fv_len - js_len + 1] = '\0';
```

Comment:
If we allocate N bytes with calloc and write at buf[N] this would write over the boundary?

Reply:
The `+ 1` on line 44 is an off by one error. The number of allocated bytes is correct, but `buf[fv_len - js_len] = '\0';` is the correct way to null-terminate that buffer.

Reply:
Also @nickva just pointed out that `calloc` zeroes out the returned memory so I'd suggest either replacing the manual null termination with a note reminding readers that `calloc` zeroes everything out. Normally, I'd also suggest switching to malloc since you're immediately touching every byte that was allocated before returning the string so having calloc zero everything out is wasted work, but given the location and purpose here that's shouldn't be an issue.

Reply:
@nickva @davisp 
You both were completely right. I'll talk my way out of it with a copy and paste error. 🙈😉
Will replace calloc with malloc.